### DISCUSSION TITLE
如何对定理环境之注（remark）自动编号？

### DISCUSSION THREAD

**Discussion by grwei** [sjtug/SJTUThesis#636] [提问] [answered]
小白提问：如何对定理环境之注（remark）自动编号，如同定义（definition）、例子（example）等等那样？谢谢。

**Comment by skyzh** [comment]
在 `sjtuthesis.cls` 中，有
```
\newtheorem*{remark}    {\sjtu@name@remark}
```

修改成
```
\newtheorem{remark}    {\sjtu@name@remark}      [chapter]
```

即可。

```
\begin{remark}
  测试
\end{remark}
```

<img width="281" alt="image" src="https://user-images.githubusercontent.com/4198311/119294435-070f9280-bc87-11eb-86f1-d74dfee1b52a.png">

**Reply by skyzh** [reply]
cc @AlexaraWu 感觉我们可以按惯例定义 `remark` 和 `remark*` 环境，前者有编号，后者无编号，让用户自己选择用哪一种（

**Reply by AlexaraWu** [reply]
模板没有必要弄得太复杂，保持基本设置就好。有特殊需求在导言区解决，尽量避免修改模板类文件。

**Reply by grwei** [reply]
问题已解决，感谢 @skyzh @AlexaraWu 的即时解答。

**Reply by jackyhum** [reply]
@skyzh @AlexaraWu 大佬们，现在2025年使用上述方法实现`注`带编号的方法在文件中找不到源码。因此我尝试在`sjtuthesis.cls`文件中，将`remark`移入有编号的列表，仍然无法实现期望的功能
`\cs_new_protected:Nn \__sjtu_new_theorems:
  {
    \clist_map_inline:nn
      {
        assumption, axiom, conjecture, corollary, definition, example,
        exercise, lemma, problem, proposition, theorem, **remark**
      }
      { \exp_args:Nnv  \newtheorem  {##1} { c__sjtu_name_ ##1 _tl } [ chapter ] }
    \clist_map_inline:nn
      { solution }
      { \exp_args:NNnv \newtheorem* {##1} { c__sjtu_name_ ##1 _tl } }
  }`

**Reply by NemoYuan2008** [reply]
@jackyhum 直接在导言区添加如下代码：

按章节编号：

```latex
\newtheorem{remark}{注}[chapter]
```

连续编号：

```latex
\newtheorem{remark}{注}
```

模板检测到你自定义了 remark 定理环境后将不会对其重定义

**Reply by jackyhum** [reply]
@NemoYuan2008 ，使用之后显示remark已经被定义，
<img width="868" alt="image" src="https://github.com/user-attachments/assets/61632a14-d55b-4f31-a5fb-34f8d81da38d" />

以`remark`全局搜索整个工程文件中，最相关的就是之前这个地方
> @skyzh @AlexaraWu 大佬们，现在2025年使用上述方法实现`注`带编号的方法在文件中找不到源码。因此我尝试在`sjtuthesis.cls`文件中，将`remark`移入有编号的列表，仍然无法实现期望的功能 `\cs_new_protected:Nn \__sjtu_new_theorems: { \clist_map_inline:nn { assumption, axiom, conjecture, corollary, definition, example, exercise, lemma, problem, proposition, theorem, **remark** } { \exp_args:Nnv \newtheorem {##1} { c__sjtu_name_ ##1 _tl } [ chapter ] } \clist_map_inline:nn { solution } { \exp_args:NNnv \newtheorem* {##1} { c__sjtu_name_ ##1 _tl } } }`

**Reply by AlexaraWu** [reply]
你用的应该是 v2.1 之前的版本，换新版的模板试试。

**Comment by AlexaraWu** [comment, marked as answer]
`ntheorem` 包的话可以使用 `\renewtheorem` 重定义定理环境。

按章节编号：
```latex
\renewtheorem{remark}{注}[chapter]
```

连续编号：
```latex
\renewtheorem{remark}{注}
```
